doc/decisions/0_drafts/readonly_keynames.md

```diff
@@ -0,0 +1,49 @@
+# Readonly Keynames
+
+## Problem
+
+It might be a good idea to prevent changes to the keyname after creation.
+Currently, key names are automatically read-only when they are contained in a keyset.
+
+There may also be situations where changing the name of a key after its creation is required.
+In some situations, it may be wise to reuse a key instead of deleting it and creating a new one.
+
+## Constraints
+
+1. `keyAddName` et al. still have to work up to a certain point
+2.
+3.
+
+## Assumptions
+
+1.
+2.
+3.
+
+## Considered Alternatives
+
+### Alternative A
```

Comment:
```suggestion
### Separate API for keynames

Use the proposed `ElektraBuffer` struct to create a separate API for keynames independent of the `Key` API.
```

Intentionally not very detailed, because the buffer decision is still in progress.